extension/data.ts

```diff
@@ -277,112 +295,96 @@ class RcxDict {
               }
               o = { word: word, type: 0xff, reason: '' } as Deinflection;
               if (have[newWord] !== undefined) {
-                o = r[have[newWord]];
+                o = possibleDeinflections[have[newWord]];
                 o.type |= rule.typeMask >> 8;
 
                 continue;
               }
-              have[newWord] = r.length;
-              if (r[i].reason.length) {
+              have[newWord] = possibleDeinflections.length;
+              if (possibleDeinflections[i].reason.length) {
                 o.reason =
-                  this.difReasons[rule.reasonIndex] + ' &lt; ' + r[i].reason;
+                  this.difReasons[rule.reasonIndex] +
+                  ' &lt; ' +
+                  possibleDeinflections[i].reason;
               } else {
                 o.reason = this.difReasons[rule.reasonIndex];
               }
               o.type = rule.typeMask >> 8;
               o.word = newWord;
-              r.push(o);
+              possibleDeinflections.push(o);
             }
           }
         }
       }
-    } while (++i < r.length);
+    } while (++i < possibleDeinflections.length);
 
-    return r;
+    return possibleDeinflections;
   }
 
-  // katakana -> hiragana conversion tables
-  ch: number[] = [
-    0x3092, 0x3041, 0x3043, 0x3045, 0x3047, 0x3049, 0x3083, 0x3085, 0x3087,
-    0x3063, 0x30fc, 0x3042, 0x3044, 0x3046, 0x3048, 0x304a, 0x304b, 0x304d,
-    0x304f, 0x3051, 0x3053, 0x3055, 0x3057, 0x3059, 0x305b, 0x305d, 0x305f,
-    0x3061, 0x3064, 0x3066, 0x3068, 0x306a, 0x306b, 0x306c, 0x306d, 0x306e,
-    0x306f, 0x3072, 0x3075, 0x3078, 0x307b, 0x307e, 0x307f, 0x3080, 0x3081,
-    0x3082, 0x3084, 0x3086, 0x3088, 0x3089, 0x308a, 0x308b, 0x308c, 0x308d,
-    0x308f, 0x3093,
-  ];
+  isKana(charCode: number): boolean {
+    return (
+      (charCode >= KANA.HIRAGANA_START && charCode <= KANA.HIRAGANA_END) ||
+      (charCode >= KANA.KATAKANA_START && charCode <= KANA.KATAKANA_END) ||
+      (charCode >= KANA.HW_KATAKANA_START && charCode <= KANA.HW_KATAKANA_END)
+    );
+  }
+  /**
+   * Returns the input string converted into hiragana. If any characters are not
```

Comment:
nit: It's good to reference the inputs expliictly in the summary statement so that you don't need to use an `@param`. The `@` markers can be useful for additional information that doesn't flow naturally in the summary statement but there's no need to repeat information there that can be read once at the beginning!

So in this case something like:
```
Returns the given `kanaWord` with katakana and half width characters converted to full-width hiragana. ヴ is not converted. If a non-kana characters is found in `kanaWord`, that and the following characters are omitted from the returned conversion.
```